mev_inspect/classifiers/utils.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional, List, Sequence
+
+from mev_inspect.schemas.swaps import Swap
+from mev_inspect.schemas.transfers import Transfer, ETH_TOKEN_ADDRESS
+
+from mev_inspect.schemas.traces import DecodedCallTrace, ClassifiedTrace
+
+
+def create_swap_from_transfers(
+    trace: DecodedCallTrace,
+    recipient_address: str,
+    prior_transfers: List[Transfer],
+    child_transfers: List[Transfer],
+) -> Optional[Swap]:
+    pool_address = trace.to_address
+
+    if recipient_address is None:
+        return None
```

Comment:
why not raise an exception? This seems like a silent failure.

Reply:
can recipient address be None? you have it type annotated as str not Optional[str]

Reply:
the recipient address can't be none as it is passed in parse_swap as an attribute of a DecodedCallTrace. This logic was already here, which i did not intend to alter and move in the same PR. This only moves the existing swap logic into the swap classifiers